language-server/lsp/lspData.ml

```diff
@@ -91,9 +104,36 @@ module ServerCapabilities = struct
   | `Int 2 -> Incremental
   | _ -> Yojson.json_error "invalid value"
 
+  type completionOptions = {
+    triggerCharacters : string list option;
+    allCommitCharacters : string list option;
+    resolveProvider : bool option;
+    completionItemLabelDetailsSupport : bool option;
+  } [@@deriving yojson]
+
+  let yojson_of_completionOptions options =
```

Comment:
Sorry, I should have explained better: you don't need to write this function anymore. The `[@@deriving yojson]` annotation you put on the type derives it automatically.

Reply:
The spec does not accept a null value: <https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#completionOptions> 
And it is my understanding yojson would decode a option to null instead of leaving it out. 
So should we remove the deriving, the values we don't use (leaving only resolveProvider) or the option and then use the correct values for each.

Reply:
You are right. We should add `[@yojson.option]` to optional fields, to strictly obey the spec.

Reply:
Do you mind adding it where appropriate? It will make `None`-valued fields be omitted rather than set to `null`.

Reply:
Yep, i will add it.